jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -156,76 +154,67 @@ public boolean isSpatRecord() {
 
 	private void publishList(XmlUtils xmlUtils, List<OdeData> dataList) throws JsonProcessingException {
 		serialId.setBundleSize(dataList.size());
+
 		for (OdeData odeData : dataList) {
 			OdeLogMetadata msgMetadata = (OdeLogMetadata) odeData.getMetadata();
 			OdeMsgPayload msgPayload = (OdeMsgPayload) odeData.getPayload();
 			msgMetadata.setSerialId(serialId);
 
 			if (isDriverAlertRecord()) {
-				logger.debug("Publishing a driverAlert.");
-
 				publisher.publish(JsonUtils.toJson(odeData, false),
 						publisher.getOdeProperties().getKafkaTopicDriverAlertJson());
+			} else if (isBsmRecord()) {
+				publisher.publish(JsonUtils.toJson(odeData, false),
+				publisher.getOdeProperties().getKafkaTopicOdeRawEncodedBSMJson());
+			} else if (isSpatRecord()) {
+				publisher.publish(JsonUtils.toJson(odeData, false),
+					publisher.getOdeProperties().getKafkaTopicOdeRawEncodedSPATJson());
 			} else {
-				if (isBsmRecord()) {
-					logger.debug("Publishing a BSM");
-				} else if (isSpatRecord()) {
-					logger.debug("Publishing a Spat");
-				} else {
-					logger.debug("Publishing a TIM or MAP");
-				}
-
-				if (!(isSpatRecord() && msgMetadata instanceof OdeSpatMetadata
-        			&& !((OdeSpatMetadata) msgMetadata).getIsCertPresent())) {
-					if (checkHeader(msgPayload) == "Ieee1609Dot2Data") {
-						Asn1Encoding msgEncoding = new Asn1Encoding("root", "Ieee1609Dot2Data", EncodingRule.COER);
-						msgMetadata.addEncoding(msgEncoding);
-					}
+				// Determine the message type (MAP or TIM are the current other options)
+				String messageType = determineMessageType(msgPayload);
+				if (messageType == "MAP") {
```

Comment:
To be consistent with isBsmRecord(), maybe consider isMapMsgType()?

Reply:
I would prefer not to make separate methods for each message type, especially since we will be reworking this to be generic in the future. Would this be alright to keep for now?